### ISSUE TITLE
Label list and note list styling should be different

### ISSUE THREAD

**Issue by nevans** [ruby/rdoc#1199]
In #1157
> * Enhance table and list styling for improved readability

First off, I love the attention that's been given to rdoc over the last several months.  It's looking pretty nice!  I'm very much looking forward to the next release. 🤩

The only new issue I can think of is this: label lists and note lists now render the exact same way.  Was that intentional?  IMO, it's useful to have the two different definition list styles, even though I'd love for one or both of them to be updated from what they were in v6.7.0.

Here's what they look like in v6.7.0:
![Screenshot 2024-11-05 at 12-33-04 rdoc 6.7.0 definition lists](https://github.com/user-attachments/assets/3d29493d-010a-45d0-b889-92d99d19b1fd)

And here's what they look like today at https://ruby.github.io/rdoc/ExampleRDoc_rdoc.html
![Screenshot 2024-11-05 at 12-34-17 rdoc new definition lists](https://github.com/user-attachments/assets/3ee25b02-d4e7-4a48-a8cf-ab2740a48104)

Obviously, we can apply our own stylesheets, but IMO they should be slightly different out-of-the-box.  The `float` style that was previously used only for "note" lists is the one that is now used for both.

_Personally_, between the two I like the default styling on "note" lists less than v6.7.0's "label" lists.  IMO, the left margin's unindentation after more than two lines of text is visually jarring.

**Comment by nevans** [comment]
If I have time, I'll try to submit a PR with updated styles for both lists soon.  But what do you think?  Do you agree that they _should_ be styled differently?  Should they just be returned to their 6.7.0 styles, or should they be updated to something (subjectively) nicer?

**Comment by drbrain** [comment]
I don't think they need to look different when presented as HTML.

These are two types of markup that allow you to choose the most convenient way to create an HTML [description list](https://html.spec.whatwg.org/multipage/grouping-content.html#the-dl-element) based on the keys.

**Comment by nevans** [comment]
@drbrain I suppose that's fair.  I can see that it's useful to have alternative representations for the exact same thing, with the only distinction that one reads more nicely in the source comments.  And I _do_ appreciate that rdoc has prioritized nice looking human-readable source comments over syntax-heavy comments-metadata.  Also, I hadn't realized (until your comments, when I tried it out and looked at the source) that `ri` _already_ formatted them both the same (and always has?).

That said, since I've mostly focused on making the HTML presentation look nice, I'd definitely conceptualized them as intentionally different in semantics, presentation, or both.  Both created description lists, but the classes and styling were different.  The classes are _still_ different, so I _can_ still apply my own styles.

Anyway, I do think that:
1. There's a subtle but useful semantic distinction between the two types.  It's not hugely important, but it's maybe nice to have.
2. It's useful to have two different description list presentations.
3. Following from 1 & 2, maybe `ri` should _also_ format them each slightly differently.  😉
4. But, maybe all of the above is less important than the goal you gave: two different ways to express the same thing, based on which is most convenient for the content.

   If we _are_ going to continue styling them the same, IMO we should use to the original "label" presentation style.  I've always found the "note" style's bouncing float left margin to be jarring, although that can be fixed!  Even if it is fixed, I still think the old "label" style is more appropriate in most contexts.  It's the style that `ri` uses for _both_ types of description list.)

I've already been adding my own stylesheet and monkey-patching `Markup::ToHtml` to override `list_type == :NOTE` and `LIST_TYPE_TO_HTML[:NOTE]` to make them even _more_ distinct (with code I originally copied from `hanna`).  When I saw the style changes from #1157 and other commits, I was going to convert my ToHtml changes into pure CSS, and update it to fit in better with the rest of the updated darkfish theme.  And obviously I can still do that.  But losing the distinction seemed like it may have been more oversight than intentional.  (Any thoughts from @st0012?)

(Side note, I'd somehow missed that HTML 5 renamed `dd` from "definition list" to "description list"... it makes sense, but I'd missed that change until reading your comment.  🙂)
